libkrun/libkruntime.c

```diff
@@ -29,15 +29,70 @@
 #define ACTUAL_CLOCK_MONOTONIC    CLOCK_MONOTONIC
 #endif
 
+#define CORE_BOUNDS_CHECK(C)                                                \
```

Comment:
I still can't help but feel these should be functions and not macros -- is there a good reason for them being macros? C macros are dangerous, so they always require more careful thought than normal functions.